src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -250,21 +251,26 @@ public void Enqueue( Event e )
         /// </returns>
         public Event Dequeue(bool blockWhenEmpty)
         {
-            lock (this.syncRoot)
+            lock (syncRoot)
             {
-                while (this.queue.Count == 0)
+                while (queue.Count == 0)
                 {
-                    if (blockWhenEmpty && !this.stopped)
+                    if (blockWhenEmpty && !stopped)
+#if NETCF
                     {
-                        Monitor.Wait(this.syncRoot);
+                        Monitor.Exit(syncRoot);
+                        syncEvent.WaitOne();
+                        Monitor.Enter(syncRoot);
+                        syncEvent.Reset();
```

Comment:
Not really.  Monitor is a static class, so extensions cannot be used.  Since lock() implicitely uses the System.Threading.Monitor class, there is no way to replace that.  I wrote three versions of this, and this was the least impactive.